lib/luaxtable.c

```diff
@@ -88,6 +192,15 @@ static const lunatik_class_t luaxtable_class = {
 	.sleep = false,
 };
 
+static inline void luaxtable_setbuffer(lua_State *L, int idx, luaxtable_t *xtable)
+{
+	lunatik_require(L, data);
+	xtable->skb = lunatik_checknull(L, luadata_new(NULL, 0, false, LUADATA_OPT_NONE));
+	lunatik_cloneobject(L, xtable->skb);
```

Comment:
why cloning the object here?

Reply:
`luadata_new` allocates memorym sets the class and returns a pointer to luadata object, right ? So it's not on the stack yet. So we need to clone it, then `lunatik_setregistry` in the next line will call : 
```c
	lua_pushvalue(L, ix);
	lua_rawsetp(L, LUA_REGISTRYINDEX, key); /* pop value */
```

Otherwise, it panics. I used `lunatik_cloneobject` since it calls lunatik_setclass, which pushes it onto the stack : 
```c
	if (luaL_getmetatable(L, class->name) == LUA_TNIL)
		luaL_error(L, "metatable not found (%s)", class->name);
	lua_setmetatable(L, -2);
	lua_pushlightuserdata(L, (void *)class);
	lua_setiuservalue(L, -2, 1); /* pop class */
```